app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -206,17 +211,431 @@ class StateFragmentTest {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+
+      clickContinueNavigationButton()
+      clickPreviousNavigationButton()
+      clickNextNavigationButton()
+
+      rotateToLandscape()
+
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "Which bird can sustain flight for long periods of time?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_moveFwd_moveFwd_moveBackwd_moveBackwd_stayOnSameState() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      selectMultipleChoiceOption(optionPosition = 2)
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+      clickPreviousNavigationButton()
+
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "What fraction represents half of something?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_moveFwdTwice_moveBackwdTwice_configChange_stayOnSameState() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      selectMultipleChoiceOption(optionPosition = 2)
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+      clickPreviousNavigationButton()
+
+      rotateToLandscape()
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "What fraction represents half of something?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
   //  7. Verifying that old answers were present when navigation backward/forward.
+  @Test
+  fun testStateFragment_loadExp_secondState_submitAnswer_moveForward_moveBackward_answerVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+
+      scrollToViewType(SUBMITTED_ANSWER)
+      onView(withId(R.id.submitted_answer_text_view)).check(
+        matches(
+          withText("1/2")
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_landscape_submitAnswer_moveForward_moveBackward_answerVisible() {
```

Comment:
```suggestion
  fun testStateFragment_loadExp_configChange_submitAnswer_moveForward_moveBackward_answerVisible() {
```

Reply:
if char exceeds 100, then cut short `fwd` and`bckwd`

Reply:
Done